staging/src/k8s.io/apiserver/pkg/registry/rest/delete.go

```diff
@@ -135,3 +136,37 @@ func BeforeDelete(strategy RESTDeleteStrategy, ctx context.Context, obj runtime.
 	}
 	return true, false, nil
 }
+
+// AdmissionToValidateObjectDeleteFunc returns a admission validate func for object deletion
+func AdmissionToValidateObjectDeleteFunc(admit admission.Interface, staticAttributes admission.Attributes) ValidateObjectFunc {
+	mutatingAdmission, isMutatingAdmission := admit.(admission.MutationInterface)
+	validatingAdmission, isValidatingAdmission := admit.(admission.ValidationInterface)
+
+	return func(old runtime.Object) error {
+		if !isMutatingAdmission && !isValidatingAdmission {
+			return nil
+		}
+		finalAttributes := admission.NewAttributesRecord(
+			nil,
+			old,
+			staticAttributes.GetKind(),
+			staticAttributes.GetNamespace(),
+			staticAttributes.GetName(),
+			staticAttributes.GetResource(),
+			staticAttributes.GetSubresource(),
+			staticAttributes.GetOperation(),
+			staticAttributes.GetUserInfo(),
+		)
+		if isMutatingAdmission && mutatingAdmission.Handles(finalAttributes.GetOperation()) {
```

Comment:
I know the original code called mutatingAdmission as well, but I don't understand why we did that. What can be mutated? We didn't even pass obj or oldObj to the mutation admission chain.

Reply:
you can reject earlier with a mutating webhook, even if you don't plan to mutate anything